connectors/source.py

```diff
@@ -115,8 +115,12 @@ class BaseDataSource:
     service_type = None
 
     def __init__(self, configuration):
+        if not isinstance(configuration, DataSourceConfiguration):
+            raise TypeError(
+                f"Configuration expected type is DataSourceConfiguration, actual: {type(configuration).__name__}."
```

Comment:
```suggestion
                f"Configuration expected type is {type(self).__name__}, actual: {type(configuration).__name__}."
```

Would make it a bit more robust to a change of the class name.

Reply:
Good call!